### PR TITLE
Bump Python and add REUSE badge

### PR DESCRIPTION
**Pull request by MoritzWeber0**

### CODE REVIEW

File: .github/workflows/deploy-docker-images.yml [hunk 1]
```diff
@@ -15,19 +15,19 @@ jobs:
...
+          - "6.0.0"
     name: Capella ${{ matrix.capella_version }}
     steps:
       - name: Checkout repository
         uses: actions/checkout@v3
       - name: Set up Python
         uses: actions/setup-python@v4.3.0
         with:
-          python-version: "3.10"
+          python-version: "3.11"
```
Comment (dominik003) [inline .github/workflows/deploy-docker-images.yml:30]:
```suggestion
          cache: pip
          cache-dependency-path: ./pyproject.toml
          python-version: "3.11"
```

File: .github/workflows/pre-commit.yml [hunk 2]
```diff
@@ -16,7 +16,7 @@ jobs:
       - uses: actions/checkout@v3
       - uses: actions/setup-python@v2
```
Comment (dominik003) [inline .github/workflows/pre-commit.yml:17]:
```suggestion
      - uses: actions/setup-python@v4
```

File: .github/workflows/pre-commit.yml [hunk 3]
```diff
@@ -16,7 +16,7 @@ jobs:
...
         with:
-          python-version: "3.10"
+          python-version: "3.11"
```
Comment (dominik003) [inline .github/workflows/pre-commit.yml:19]:
```suggestion
          cache: pip
          cache-dependency-path: ./pyproject.toml
          python-version: "3.11"
```

**Review by dominik003 — APPROVED** [review]
Looks good, there are just a few small improvements that could be added.

File: .github/workflows/deploy-docker-images.yml (hunk 1 above)
Comment (MoritzWeber0) [reply .github/workflows/deploy-docker-images.yml:30]: Done.

File: .github/workflows/pre-commit.yml (hunk 3 above)
Comment (MoritzWeber0) [reply .github/workflows/pre-commit.yml:19]: Here, it doesn't make sense (as it only installs `pre-commit` and the `pyproject.toml` is not the right cache key.

Comment (MoritzWeber0) [reply .github/workflows/pre-commit.yml:19]: But I've added `cache: pip`

Comment (MoritzWeber0) [reply .github/workflows/pre-commit.yml:20]: Hmm, doesn't seem to work: `No file in /home/runner/work/capella-dockerimages/capella-dockerimages matched to [**/requirements.txt], make sure you have checked out the target repository`. Removed it again.

**Pull request merged**